turbolinks/src/main/java/com/basecamp/turbolinks/TurbolinksJavascriptInjector.java

```diff
@@ -0,0 +1,74 @@
+package com.basecamp.turbolinks;
+
+import android.content.Context;
+import android.util.Base64;
+import android.webkit.ValueCallback;
+import android.webkit.WebView;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class TurbolinksJavascriptInjector {
+    private static String scriptInjectionFormat = "(function(){var parent = document.getElementsByTagName('head').item(0);var script = document.createElement('script');script.type = 'text/javascript';script.innerHTML = window.atob('%s');parent.appendChild(script);return true;})()";
+
+    // ---------------------------------------------------
+    // Public
+    // ---------------------------------------------------
+
+    /**
+     * <p>Injects Javascript into the webView.</p>
+     *
+     * @param turbolinksSession The TurbolinksSession.
+     * @param context           Any Android context.
+     * @param webView           The shared webView.
+     * @param filePath          Local file path relative to the main/src directory.
+     */
+    public static void injectJavascript(final TurbolinksSession turbolinksSession, Context context, WebView webView, String filePath) {
+        try {
+            String script = TurbolinksJavascriptInjector.getContentFromAssetFile(context, filePath);
+            String jsCall = String.format(scriptInjectionFormat, script);
+
+            webView.evaluateJavascript(jsCall, new ValueCallback<String>() {
+                @Override
+                public void onReceiveValue(String s) {
+                    if (turbolinksSession != null) {
+                        turbolinksSession.turbolinksBridgeInjected = Boolean.parseBoolean(s);
```

Comment:
This block served a specific purpose of setting a flag in the TurbolinksSession, specific to the turbolinks_bridge.js being successfully injected. This would need to be set only for when that was injected properly.

Also, the caller of this method would probably want some kind of asynch callback to know its state, so you'd want a way to handle that.